File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    const sieve = [];
    
    for (let k = 2; k <= n; k++) {
        sieve[k] = k;
    }
    
    for (let i = 2; i <= n; i++) {
        if (sieve[i] === false) continue;
        
        for (let j = i + i; j <= n; j += i) {
            sieve[j] = false;
        }
    }
    
    return sieve.filter((el) => el).length;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def solution(numbers):
    numbers_set = set()
    
    for i in range(1, len(numbers) + 1):
        perms = permutations(numbers, i)
        numbers_set.update([int(''.join(p)) for p in perms])
        
    return len(list(filter(lambda x: is_prime(x),numbers_set)))
        
        
        
def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True